test/extensions/filters/udp/dns_filter/dns_filter_fuzz_test.cc

```diff
@@ -0,0 +1,68 @@
+#include "common/common/logger.h"
+
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "test/fuzz/fuzz_runner.h"
+#include "test/fuzz/utility.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+namespace {
+
+const std::string generateQuery(FuzzedDataProvider* data_provider) {
+  size_t query_size = data_provider->ConsumeIntegralInRange<size_t>(0, 512);
+  return data_provider->ConsumeRandomLengthString(query_size);
+}
+
+DEFINE_FUZZER(const uint8_t* buf, size_t len) {
+  FuzzedDataProvider data_provider(buf, len);
+  const bool recurse = data_provider.ConsumeIntegralInRange<bool>(0, 1);
+  const uint16_t retry_count = data_provider.ConsumeIntegralInRange<uint16_t>(0, 3);
+
+  NiceMock<Runtime::MockRandomGenerator> random;
```

Comment:
when running the fuzz target how was the speed? if low, consider making this static

Reply:
As far as speed goes, the fuzz test by itself takes '0" seconds:
```
Done 1000 runs in 0 second(s)
```
I removed the expectation which significantly slowed it down (as you correctly pointed out below).

Coverage output is:
```
Overall coverage rate:
  lines......: 2.3% (960 of 41603 lines)
  functions..: 2.0% (138 of 6943 functions)
Code coverage 2.3 is lower than limit of 97.0
```

Reply:
Thanks! Usually you have to run the fuzzer for O(10) seconds (without the -runs flag, if you want you can set -max_total_time=60 for a minute) and then it'll eventually stabilize and print the real speed.

About coverage -- total coverage is expected low, I just meant if it's possible to view the report and see the DNS parser coverage.
`FUZZ_COVERAGE=true VALIDATE_COVERAGE=false test/run_envoy_bazel_coverage.sh ${FUZZ_TEST}_with_libfuzzer` will make it not fail and print out the report

Reply:
Spoke on slack. Filter fuzz coverage is 30.3%.  Took an action item to try to improve this.